app/src/main/java/com/example/soptseminar1/presentation/profile/repository/RepositoryAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.soptseminar1.presentation.profile.repository
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.soptseminar1.data.model.response.ResponseGithubUserRepo
+import com.example.soptseminar1.databinding.RepoLayoutBinding
+
+class RepositoryAdapter : ListAdapter<ResponseGithubUserRepo, RepositoryAdapter.RepositoryViewHolder>(
+    diffUtil
+) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepositoryViewHolder {
+        val binding = RepoLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return RepositoryViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: RepositoryViewHolder, position: Int) {
+        holder.onBind(currentList[position])
+    }
+
+    class RepositoryViewHolder(
+        private val binding: RepoLayoutBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(data: ResponseGithubUserRepo) {
+            binding.repoName.text = data.name
+            binding.repoDescription.text = data.description
+        }
+    }
+
+    companion object {
+        val diffUtil = object : DiffUtil.ItemCallback<ResponseGithubUserRepo>(){
+            override fun areItemsTheSame(
+                oldItem: ResponseGithubUserRepo,
+                newItem: ResponseGithubUserRepo
+            ): Boolean {
+                return oldItem.name == newItem.name || oldItem.description == newItem.description
```

Comment:
```suggestion
                return oldItem.name == newItem.name
```

아까는 제대로 제가 이해를 못 했어서 왜 조건을 2개를 걸었는지 물어봤는데, onBind() 메서드를 보고 왜 2개를 썼는지 이해가 되어 리뷰를 드립니다. 기본적인 DiffUtil의 동작방식은 이렇습니다.

> `areItemsTheSame` 메서드를 통해 두 아이템의 Key가 같은지 확인 -> 같다면 `areContentsTheSame` 메서드를 통해 내부 내용물도 같은지 확인해 같은 아이템인지 다른 아이템인지 판별 || 다르다면 두 아이템은 다른 아이템으로 인지

이런 원리로 DiffUtil이 동작하기 때문에, `areItemsTheSame` 메서드는 두 아이템을 비교할 고유한 Key로 비교하는 것이 맞습니다. 꼭 저렇게 조건을 2개를 걸어서 `areItemsTheSame`에서 전부 걸러버릴 필요는 없다는 것입니다.